providers/azure/client/api/api.go

```diff
@@ -56,12 +61,20 @@ func CheckResponse(res *http.Response) error {
 			}
 			jerr.Error.Body = string(slurp)
 			jerr.Error.URL = res.Request.URL.String()
-			return jerr.Error
+			e = jerr.Error
+		}
+	} else {
+		e = &Error{
+			StatusCode: res.StatusCode,
+			Body:       string(slurp),
```

Comment:
old issue: slurp is nill here. Shall we use res.Status instead?

Reply:
Fixed